### ISSUE TITLE
`Process::setEnv()` is tagged as private

### ISSUE THREAD

**Issue by sanmai** [composer/xdebug-handler#66]
Whole `Process` class and specifically `Process::setEnv()` are tagged as `@private`. So far I had to [reimplement `setEnv`](https://github.com/infection/infection/pull/329/files#diff-0bd8be84061bbe7b5e24c1805d198de9R71), though in a limited fashion. 

Would it make sense to allow outside usage of this function?

**Comment by johnstevenson** [comment]
Quite probably, but I am reluctant to open up too much too soon, at this stage. However, I'm looking at implementing what you need this for in a separate helper class, so it shouldn't be an issue.

**Comment by theofidry** [comment]
Back on the issue: `Process::setEnv` is now public but still tagged as `@internal`. Are you considering removing it? Looking at the methods of `Process` they look like well defined static methods hence it shouldn't be too complicated to change them later on if necessary.

I'm however maybe just misusing it, you tell me: https://github.com/humbug/box/blob/8a56b4d/src/Console/Php/PhpSettingsHandler.php#L62-L63

**Comment by johnstevenson** [comment]
You can use `setPersistent()` in the `requiresRestart` method to set up the environment for the new process. This will keep xdebug out of any subsequent subprocesses, which I presume is your intention.

You only need to make your memory_limit adjustments once (ie you don't have to differentiate between the first and restarted processes), because the setting will get picked up and added to the tmp ini.

**Comment by theofidry** [comment]
Good point regarding the memory limit; will try out the setPersistent ASAP

**Comment by theofidry** [comment]
Worked like a charm, thanks @johnstevenson
